api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Proxy-WASM ABI.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+//
+//  ABI calls from the host into the VM.
+//
+//  These will typically be implemented by a language specific SDK which will provide an API on top
+//  of this ABI e.g. the C++ SDK provides a proxy_wasm_api.h implemenation of the API on top of this
+//  ABI.
+//
+
+// Non-stream calls.
+
+/**
+ * Called when the VM starts for each plugin.
+ * @param root_context_id is an identifier with the lifetime of the VM which will be used for
+ * one or more related plugins and their corresponding proxy_on_configure(), proxy_on_done() and
+ * proxy_on_delete() calls. It is also be provided during creation of stream context for those
+ * plugins.
+ * @param configuration_size is the size of any configuration available via proxy_get_configuration.
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+extern "C" uint32_t proxy_on_start(uint32_t root_context_id, uint32_t configuration_size);
+/**
+ * Can be called to validate a configuration (e.g. by an xDS provider) both before proxy_on_start()
+ * to verify the VM configuration or after proxy_on_start() to verify a plugin configuation.
+ * @param root_context_id is a unique identifier for the configuration verification context.
+ * @param configuration_size is the size of any configuration available via
```

Comment:
Don't understand how configuration_size is used here or what this actually does. Is this a stub which data follows this call somehow?

Reply:
See above. There is a call proxy_get_configuration which will retrieve the whole configuration.   In general the ABI passers only primitive types and the code in the VM retrieves any large objects.  We also pass in the size (e.g. of the configuration, buffer, number of headers, etc.) so that the plugin can make an informed decision to retrieve the larger object or part of whole.

Reply:
Per my comment above, does WASM require this? This seems very convoluted for no real reason.

Reply:
See above.  Users will not be dealing with the ABI.  It is an interface for SDK developers.

Reply:
Resolved below.